flowmachine/flowmachine/core/model_result.py

```diff
@@ -136,17 +141,29 @@ def to_sql(self, name: str, schema: Union[str, None] = None) -> Future:
         schema : str, default None
             Name of an existing schema. If none will use the postgres default,
             see postgres docs for more info.
+        store_dependencies : bool, default False
+            Required for compatibility with query.to_sql
+            store_dependencies=True raises NotImplementedError
 
         Returns
         -------
         Future
             Future object, containing this query and any result information.
+        
+        Raises
+        ------
+        NotImplementedError
+            If store_dependencies = True
 
         Notes
         -----
 
         This method will return a Future immediately.
         """
+        if store_dependencies:
+            raise NotImplementedError(
+                "ModelResult does not support storing dependencies."
```

Comment:
Why not?

Reply:
`ModelResult.store()`/`ModelResult.to_sql()` stores a pandas DataFrame which is the result of running a model. Since the result is already calculated, there's no efficiency benefit from storing the dependencies when storing the result.

Reply:
Right, but the model will generally speaking use the results of some queries in calculation, and indeed, you should be able to see that link if you view the graph of any model result.

Reply:
True. I guess the reason I'm slightly conflicted is that when `store_dependencies=True` we store only the necessary dependencies (i.e. we don't store queries that are only dependencies of already-stored queries). In this case the ModelResult isn't _stored_, but it has been _run_, so I can't decide whether it's more consistent to store the dependencies or not to.

Reply:
On the other hand, the dependencies will already have been stored when the model was run, so unless they've been removed from the cache it doesn't make any difference whether or not we try to store them again. So I suppose it doesn't make much sense to disallow setting `store_dependencies=True`.